src/dataExplorer/context/columns.tsx

```diff
@@ -0,0 +1,146 @@
+import React, {
+  FC,
+  createContext,
+  useContext,
+  useMemo,
+  useState,
+  useCallback,
+} from 'react'
+
+// Contexts
+import {QueryContext, QueryScope} from 'src/shared/contexts/query'
+
+// Constants
+import {
+  CACHING_REQUIRED_END_DATE,
+  CACHING_REQUIRED_START_DATE,
+} from 'src/utils/datetime/constants'
+import {DEFAULT_LIMIT} from 'src/shared/constants/queryBuilder'
+
+// Types
+import {Bucket, RemoteDataState} from 'src/types'
+
+// Utils
+import {
+  IMPORT_REGEXP,
+  IMPORT_STRINGS,
+  IMPORT_INFLUX_SCHEMA,
+  SAMPLE_DATA_SET,
+  FROM_BUCKET,
+} from 'src/dataExplorer/shared/utils'
+
+interface Props {
+  scope: QueryScope
+}
+
+interface ColumnsContextType {
+  columns: string[]
+  loading: RemoteDataState
+  getColumns: (bucket: Bucket, measurement: string) => void
+  resetColumns: () => void
+}
+
+const DEFAULT_CONTEXT: ColumnsContextType = {
+  columns: [],
+  loading: RemoteDataState.NotStarted,
+  getColumns: (_b: Bucket, _m: string) => {},
+  resetColumns: () => {},
+}
+
+export const ColumnsContext = createContext<ColumnsContextType>(DEFAULT_CONTEXT)
+
+export const ColumnsProvider: FC<Props> = ({children, scope}) => {
+  // Context
+  const {query: queryAPI} = useContext(QueryContext)
+
+  // States
+  const [columns, setColumns] = useState<string[]>([])
+  const [loading, setLoading] = useState<RemoteDataState>(
+    RemoteDataState.NotStarted
+  )
+
+  const getColumns = useCallback(
+    async (bucket: Bucket, measurement: string) => {
+      if (!bucket || !measurement) {
+        return
+      }
+
+      setLoading(RemoteDataState.Loading)
+
+      // Simplified version of query from this file:
+      //   src/flows/pipes/QueryBuilder/context.tsx
+      // Note that sample buckets are not in storage level.
+      //   They are fetched dynamically from csv.
+      //   Here is the source code for handling sample data:
+      //   https://github.com/influxdata/flux/blob/master/stdlib/influxdata/influxdb/sample/sample.flux
+      //   That is why _source and query script for sample data is different
+      let _source = IMPORT_REGEXP
+      if (bucket.type === 'sample') {
+        _source += SAMPLE_DATA_SET(bucket.id)
```

Comment:
Yeah, they are very similar but slightly different. The query text in [groupKeys.tsx](https://github.com/influxdata/ui/blob/master/src/dataExplorer/context/groupKeys.tsx#L99-100) doesn't filter out `_start` and `_stop`

I thought of centralize them. The reason why I didn't end of doing that was because of the [rule of three](https://erikbern.com/2017/08/29/the-software-engineering-rule-of-3.html). I'm open to centralize them if you feel strongly!